mantidproject/mantid
# SCD algorithm for finding peaks and UB matrix

**Description of work.**

Adds a new algorithm which takes DEMAND data in Q-space and creates a `PeaksWorkspace` with peaks and a new calculated UB matrix refined by a given cell type and centering. The UB matrix can also be calculated from lattice parameters if provided. 

**To test:**
Make sure the unit tests `HB3AFindPeaksTest` pass.
Additionally, the following script can be run to verify peaks are being found correctly by using SliceViewer:
```
scans = [34, 182, 183, 184]
files = ','.join("/HFIR/HB3A/IPTS-24655/shared/autoreduce/HB3A_exp0724_scan{:04}.nxs".format(scan) for scan in scans)
merged_data = HB3AAdjustSampleNorm(Filename=files, MergeInputs=True)

peaks = HB3AFindPeaks(InputWorkspace=merged_data, CellType="Tetragonal", Centering="P")
```

*There is no associated issue.*

The version of this into `ornl-next` is #29936 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
